python/ctsm/test/test_unit_modify_singlept_site_neon.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+"""
+Unit tests for modify_singlept_site_neon
+
+You can run this by:
+    python -m unittest test_unit_modify_singlept_site_neon.py
+"""
+
+import os
+import shutil
+import sys
+import tempfile
+import unittest
+from datetime import date
+import xarray as xr
+
+# -- add python/ctsm  to path (needed if we want to run the test stand-alone)
+_CTSM_PYTHON = os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir, os.pardir)
+sys.path.insert(1, _CTSM_PYTHON)
+
+# pylint: disable=wrong-import-position
+from ctsm import unit_testing
+from ctsm.site_and_regional.modify_singlept_site_neon import (
+    get_neon,
+    find_surffile,
+    update_metadata,
+    update_time_tag,
+    check_neon_time,
+)
+
+# pylint: disable=invalid-name
+
+
+class TestModifySingleptSiteNeon(unittest.TestCase):
+    """
+    Basic class for testing modify_singlept_site_neon.py.
+    """
+
+    def setUp(self):
+        """
+        Make /_tempdir for use by these tests.
+        Check tempdir for history files
+        """
+        self._tempdir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        """
+        Remove temporary directory
+        """
+        shutil.rmtree(self._tempdir, ignore_errors=True)
+
+    def test_get_neon(self):
+        """
+        Test to see if neon data for valid site name is found
+        """
+        site_name = "ABBY"
+        neon_dir = self._tempdir
+        file = get_neon(neon_dir, site_name)
+        self.assertEqual(file.split("/")[-1][:4], "ABBY", "CSV file did not download as expected")
+
+    def test_get_neon_false_site(self):
+        """
+        Test to see if neon data for invalid site name is found
+        """
+        site_name = "ABY"
+        neon_dir = self._tempdir
+        with self.assertRaises(SystemExit):
+            get_neon(neon_dir, site_name)
+
+    def test_find_surffile(self):
+        """
+        Test that surface file does not exist in tempdir and raises system exit error
+        """
+        surf_dir = self._tempdir
+        site_name = "BART"
+        pft_16 = True
+        with self.assertRaises(SystemExit):
+            find_surffile(surf_dir, site_name, pft_16)
+
+    # TODO: need to fix unkown file type error?
+    # def test_find_soil_structure(self):
+    #    """
+    #    Test to ensure that correct attributes are found for find_soil_structure.
+    #    soil_texture_raw_data_file_name should be found, and test should go through sysexit.
+    #    """
+    #    surf_file = "testinputs/surfdata_1x1_mexicocityMEX_hist_16pfts_Irrig_CMIP6_simyr2000_c221206.nc"
+    #    f1 = xr.open_dataset(surf_file)
```

Comment:
Also may be worth confirming that `test_1x1_mexicocity` still passes.

Reply:
Good idea. Thanks for pointing out that this test also uses the 1x1 Mexico City file.

It looks like the `test_1x1_mexicocity` within `test_sys_fsurdat_modifier.py` is not working. 

I'm getting the same error as what I was seeing above:
`OSError: [Errno -51] NetCDF: Unknown file format: b'/glade/work/tking/ctsm/python/ctsm/test/testinputs/surfdata_1x1_mexicocityMEX_hist_16pfts_Irrig_CMIP6_simyr2000_c221206.nc'`

Note that I did add the following lines in order to test with unittest:
`# -- add python/ctsm  to path (needed if we want to run the test stand-alone)
_CTSM_PYTHON = os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir, os.pardir)
sys.path.insert(1, _CTSM_PYTHON)`

I ran the test from within the `ctsm_pylib` environment with `python -m unittest test_sys_fsurdat_modifier.py`.

A number of other tests within `test_sys_fsurdat_modifier` also failed-- we should probably revisit that testing script, too.

@ekluzek had you run these tests in a different way than how I describe above in order to correctly find/open the netcdf files?

Reply:
Hi @TeaganKing this is a "git lfs" issue. I could tell because when you can that NetCDF file it is actually a text file that gives the LFS info for that file.

So go through the directions in the 

/glade/work/tking/ctsm/python/ctsm/test/testinputs/README.md

and hopefully that will get you going. If not let's meet to get you going. This should all be resolved after you get that figured out. Unfortunately this is one of those things where the errors you see don't make the actual problem obvious. It would be good to improve that...

Reply:
Oh, got it! That makes a lot of sense that I don't actually have the files locally if they are from git lfs. I'll check out those directions. Thanks @ekluzek !

Reply:
With LFS, the unit tests for `modify_singlept_site_neon` are now working! I decided to just test that the file was modified today instead of changed, because if the file was changed more than once, it would fail a test that checked that the file hadn't been updated today.

@ekluzek , regarding `test_sys_fsurdat_modifier`, there does still seem to be a test failure on "fsurdat_out is specified in both the command line and the config file, pick one". If these system tests are being tested in another manner, other than with `python unitttest`, do they still run into this issue?

Reply:
@TeaganKing this may answer your last question...
Run the full suite of tests from the /python directory:
`./run_ctsm_py_tests`
Try it with --help first to see available options.

Reply:
LFS issue that was resolved.